nrespipe/utils.py

```diff
@@ -570,3 +574,39 @@ def get_mag_from_simbad(target_name : str):
         mag = np.nan
 
     return mag
+
+
+def download_from_s3(frameid, output_directory):
+    url = f'{settings.ARCHIVE_FRAME_URL}/{frameid}'
+    response = requests.get(url, headers=settings.ARCHIVE_AUTH_TOKEN, stream=True).json()
+    with open(os.path.join(output_directory, response['filename']), 'wb') as f:
```

Comment:
I actually don't believe this will work. Since you're using a context manager here, your tempfile will be cleaned up directly after it is written. There will be no file at that path when the function returns. The way I am thinking of changing this is to use a context manager in the caller, and pass the file object to `download_to_s3`, rather than passing it a path.

Reply:
See https://docs.python.org/3/library/tempfile.html#examples